doc_src/history.txt

```diff
@@ -36,12 +37,18 @@ These flags can appear before or immediately after one of the sub-commands liste
 
 - `-e` or `--exact` searches or deletes items in the history that exactly match the specified text string. This is the default for the `--delete` flag. Note that the match is case-insensitive by default. If you really want an exact match, including letter case, you must use the `-C` or `--case-sensitive` flag.
 
+- `--file` changes the current history file. That option can be used both independent of any other option and in conjunction with the other options. In both cases the change to the current history file is persistent and affects subsequent commands. The switch to a new history file occurs before any other option is acted on and an implicit `history --save` occurs before the switch.
+
+The "history_file_identifier" can be a simple word or a relative or absolute path name. A simple word (i.e., one without slashes) is interpreted as a file name relative to `$XDG_DATA_HOME/fish` (`$HOME/.local/share/fish` by default). If "history_file_identifier" is the special word "fish" the default history path name will be substituted. The special word "none" results in no history file being used. This is analogous to the incognito mode provided by web browsers like Chrome. A relative path (one that contains a slash but does not begin with a slash) is interpreted relative to `$PWD` at the time the command is run. The resulting path must refer to a regular file (or not exist). Special file names like /dev/null are not allowed.
```

Comment:
Symlinks are meant to be invisible and thus valid. What matters is what the link refers to. The intent is that `stat()` is used not `lstat()`. Thus returning information about the object pointed to by the symlink.

I like the idea of using "session identifiers" rather than strings which may be relative or absolute paths. Using that idiom means `read --mode-name=$name` will continue to behave as it does today. Although possibly with some tweaks to disallow characters like slashes in the identifier. My original thinking was that interpreting the value as a path provided more flexibility. An absolute path such as */dev/null* would be exactly that. Whereas a relative path would be relative to the `$XDG_DATA_HOME` root. But that is going to cause more confusion than value. Better to interpret the value as a session ID as you propose, @faho, and allow a few special values such as "fish" (for the default file) and "none" or the empty string to mean no history should be used.